tests: tests/application_development/ram_context_for_isr : build fails on several NXP platfroms

### Describe the bug

Below platforms build failure: 
- frdm_mcxc444/mcxc444
- frdm_mcxc242/mcxc242 
- frdm_mcxn947/mcxn947/cpu0/qspi 
- hexiwear/mkw40z4 
- frdm_kw41z/mkw41z4
- frdm_kl25z/mkl25z4

this failure is introduced since this application merged.


### Regression

- [ ] This is a regression.

### Steps to reproduce

```
scripts/twister -p frdm_mcxc444 -T tests/application_development/ram_context_for_isr --build-only
```
or
```
west build -b frdm_mcxc444 tests/application_development/ram_context_for_isr
```

### Relevant log output

```shell
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/tests/application_development/ram_context_for_isr/application_development.ram_context_for_isr/zephyr/zephyr_pre0.elf --intlist-section .intList --intlist-section intList --sw-isr-table --vector-tablegen_isr_tables.py: error: multiple registrations at table_index 31 for irq 31 (0x1f)Existing handler 0x20000021, new handler 0x1d9dHas IRQ_CONNECT or IRQ_DIRECT_CONNECT accidentally been invoked on the same irq multiple times?
ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

--- a/tests/application_development/ram_context_for_isr/testcase.yaml
+++ b/tests/application_development/ram_context_for_isr/testcase.yaml
@@ -13,3 +13,9 @@ tests:
       - mps4/corstone320/fvp
       - imx943_evk/mimx94398/m33
       - imx943_evk/mimx94398/m33/ddr
+      - frdm_mcxc444/mcxc444
+      - frdm_mcxc242/mcxc242
+      - frdm_mcxn947/mcxn947/cpu0/qspi
+      - hexiwear/mkw40z4
+      - frdm_kw41z/mkw41z4
+      - frdm_kl25z/mkl25z4

